interim-jan-2021/resolver-behavior-self-description.md

```diff
@@ -0,0 +1,30 @@
+# Resolver behavior self-description
+
+## Defining local-only namespaces
+* Express namespaces which only this resolver can resolve
+    * Authoritatively if the namespace collides with any global names
+* Ex 1: Enterprise resolvers serving corporation-specific namespaces
+* Ex 2: Public Wi-Fi or cellular networks offering network-local services
+
+## Defining per-namespace optimization
+* Express namespaces for which this resolver provides preferable resolutions
+* Ex 1: ISP routes public content requests to network caches
+* Ex 2: Public resolver designated to serve a namespace to limit parties privy to resolution
+
+## Defining resolver identity
+* Express information consumable by humans describing the resolver's identity
+* Ex 1: Provide human-legible documentation
+    * Most likely a web page link to explain server identity, terms of use, etc.
+	* Not used for decision making by any protocol peer; communicated to clients for display to users
```

Comment:
Several WG members including you opposed any interaction with the end user. The argument was an average end user would not understand the privacy terms, server identity and encrypted DNS.

I have started revising draft https://github.com/tireddy2/DNS-resolver-information/blob/master/draft-add-server-selection-information.txt accordingly :)

Reply:
From my perspective, the "designated resolver" upgrade scenario should be completely non-interactive.  I think these slides are about supporting other scenarios.  Some uses I can think of:
* Inspecting and debugging my current configuration
* Browsing a directory of public resolvers

Reply:
> Several WG members including you opposed any interaction with the end user

Absolutely correct, and I still feel that way.

> The argument was an average end user would not understand the privacy terms, server identity and encrypted DNS.

And this will always continue to be true IMO. Please don't stop your draft direction change :) unless what I'm saying below clarifies my previous "get policy out of my upgrade process" verbiage.

I agree with Ben: this doesn't mean "make the upgrade process interactive" to me. Rather, this is a way to provide "other info about this resolver intended for humans" to be considered by an auditor or debugger. 

Example: if there was a standard way to return a documentation link, Google DoH servers could return [this page](https://developers.google.com/speed/public-dns/docs/doh/). It would be nice as a client implementor if there was a standard way to retrieve per-resolver documentation.

Basically, this is about returning man pages that might include terms of server use, not about returning EULAs which require an agreement to the terms of use. The pivot point is about it being a part of the encryption or upgrade negotiation or not. I am definitely opposed to this link being considered a passive agreement (or being made a requirement for active agreement) to terms of use on the server. 

Does this help?

Reply:
> 
> 
> From my perspective, the "designated resolver" upgrade scenario should be completely non-interactive. I think these slides are about supporting other scenarios. 

Okay.

> Some uses I can think of:
> 
>     * Inspecting and debugging my current configuration

Are you referring to OS/IT admin inspecting and debugging the current configuration for troubleshooting ?

> 
>     * Browsing a directory of public resolvers

I think you mean browsing a directory DoH URI services. But browsing requires the end user to iterate through the list of services to make a selection.

Reply:
> 
> 
> > Several WG members including you opposed any interaction with the end user
> 
> Absolutely correct, and I still feel that way.
> 
> > The argument was an average end user would not understand the privacy terms, server identity and encrypted DNS.
> 
> And this will always continue to be true IMO. Please don't stop your draft direction change :) unless what I'm saying below clarifies my previous "get policy out of my upgrade process" verbiage.
> 
> I agree with Ben: this doesn't mean "make the upgrade process interactive" to me. Rather, this is a way to provide "other info about this resolver intended for humans" to be considered by an auditor or debugger.
> 
> Example: if there was a standard way to return a documentation link, Google DoH servers could return [this page](https://developers.google.com/speed/public-dns/docs/doh/). It would be nice as a client implementor if there was a standard way to retrieve per-resolver documentation.
> 
> Basically, this is about returning man pages that might include terms of server use, not about returning EULAs which require an agreement to the terms of use. The pivot point is about it being a part of the encryption or upgrade negotiation or not. I am definitely opposed to this link being considered a passive agreement (or being made a requirement for active agreement) to terms of use on the server.
> 
> Does this help?

Yes, Thanks. We are in the same page :)